java/feathub-connectors/flink-connectors/flink-connector-prometheus/src/main/java/com/alibaba/feathub/flink/connectors/PrometheusSinkFunction.java

```diff
@@ -0,0 +1,342 @@
+ *       same set of keys.
+ *   <li>Map Column: Each entry in the map is emitted as a separate metric. Users must specify
+ *       exactly one label set, which must have at least one label with the value "null". The label
+ *       with value "null" will be replaced with the map's key when emitting the metric.
+ * </ul>
+ *
+ * <p>Below is an example illustrating how to define a Prometheus Sink table:
+ *
+ * <pre>
+ *     TableDescriptor.forConnector("prometheus")
+ *         .schema(
+ *             Schema.newBuilder()
+ *                 .column(
+ *                     "metric_map",
+ *                     DataTypes.MAP(DataTypes.STRING(), DataTypes.INT()))
+ *                 .withComment("feature=a,count_map_key=null")
+ *                 .column(
+ *                     "metric_array",
+ *                     DataTypes.ARRAY(DataTypes.INT()))
+ *                 .withComment("feature=a,type=int;feature=b,type=int")
+ *                 .column("metric", DataTypes.DOUBLE())
+ *                 .withComment("type=double")
+ *                 .build())
+ *         .option("serverUrl", "localhost:9091")
+ *         .option("jobName", "exampleJob")
+ *         .option("extraLabels", "k1=v1,k2=v2")
+ *         .build();
+ * </pre>
+ */
+public class PrometheusSinkFunction extends RichSinkFunction<RowData> {
+
+    private final String jobName;
+    private final Map<String, String> extraLabels;
+    private final List<ColumnInfo> columnInfos;
+    private final String serverUrl;
+    private final Boolean deleteOnShutdown;
+    private CollectorRegistry registry;
+    private PushGateway pushGateway;
+    private List<MetricUpdater<?>> metricUpdaters;
+
+    public PrometheusSinkFunction(ReadableConfig config, ResolvedSchema schema) {
+        this.jobName = config.get(JOB_NAME);
+        this.extraLabels = parseLabels(config.get(EXTRA_LABELS));
+        this.serverUrl = config.get(SERVER_URL);
+        this.deleteOnShutdown = config.get(DELETE_ON_SHUTDOWN);
+
+        this.columnInfos = new ArrayList<>();
+        for (Column column : schema.getColumns()) {
+            this.columnInfos.add(ColumnInfo.fromColumn(column));
+        }
+    }
+
+    @Override
+    public void open(Configuration parameters) throws Exception {
+        super.open(parameters);
+        registry = new CollectorRegistry();
+        pushGateway = new PushGateway(serverUrl);
+
+        metricUpdaters = new ArrayList<>();
+        for (final ColumnInfo columnInfo : this.columnInfos) {
+            metricUpdaters.add(columnInfo.getMetricWrapper(registry));
+        }
+    }
+
+    @Override
+    public void invoke(RowData row, Context context) throws Exception {
+        for (int i = 0; i < columnInfos.size(); i++) {
+            setMetric(row, i);
+        }
+        pushGateway.pushAdd(registry, jobName, extraLabels);
+    }
+
+    public void setMetric(RowData row, int index) throws IOException {
```

Comment:
Would it be better to rename this method `updateMetric` for consistency with e.g. `MetricUpdater`?